---
title: "Differential Expression Analysis"
subtitle: "_Aspergillus fumigatus conidia_ infection of airway epithelial cells"
author: "Pablo Salazar-Méndez"
date: "2024-06-22"
format:
  pdf:
    pdf-engine: xelatex
    include-in-header:
      text: |
        \usepackage{fvextra}
        \DefineVerbatimEnvironment{Highlighting}{Verbatim}{breaklines,commandchars=\\\{\}}

bibliography: references.bib
csl: apa.csl
---

# Introduction

_Aspergillus fumigatus_ is a ubiquitous fungus and opportunistic pathogen that causes serious infections in people with weakened immune systems. The interaction between its inhaled conidia and respiratory epithelial cells is a critical step in the infectious process. Understanding the molecular changes in these cells after contact provides insight into the pathogenesis of aspergillosis and identifies new therapeutic targets (@dagenais-2009). 

In this report, I analyzed RNA-seq data from primary human bronchial epithelial (HBEC) cells following interaction with _A. fumigatus conidia_ at 2, 6, and 12 hours. The goal is to identify differentially expressed involved in the host response to infection.

# Methodology

To fulfill the objectives of this project, I performed an analysis using `limma` and `recount3` R packages. The analysis was conducted on the `recount3` datasets, which provides access to a large collection of RNA-seq data. The `limma` package was used to identify differentially expressed genes between the conditions of interest.

To select a dataset I first explored some of the available projects, I runned the following command:

```{r Importing libraries}
#| echo: true
#| eval: true
#| message: false

if (!requireNamespace("BiocManager", quietly = TRUE)) {
    install.packages("BiocManager")
}

if (!requireNamespace("recount3", quietly = TRUE)) {
    BiocManager::install("recount3")
}

options(recount3_url = "https://data.idies.jhu.edu/recount3/data/")

invisible(library(recount3))
```

```{r Exploring available projects}
#| echo: true
#| eval: true
#| message: false

## Selecting a project of interest
human_projects <- available_projects()
project_name <- "SRP048565"
project_info <- subset(
    human_projects,
    project == project_name & project_type == "data_sources"
)
```
```{r Exploring the selected project}
#| echo: true
#| eval: true
#| message: true

## Cleaning unused variables
rm(human_projects, project_name)
invisible(gc())

project_info
```

# Analysis

The project `SRP048565` was selected for the analysis, which contains 18 samples from primary human bronchial epithelial (HBEC) cells following interaction with _A. fumigatus conidia_ at 2, 6, and 12 hours in order to examine the infection of airway epithelial cells at multiple time points of co-incubation (see the [NCBI site](https://www.ncbi.nlm.nih.gov/sra/SRX718212%5Baccn%5D)).

```{r Creating RSE object}
#| echo: true
#| eval: true
#| message: false

## Generating the object for the selected project
rse_SRP048565 <- create_rse(project_info)
```
```{r Exploring the object}
#| echo: true
#| eval: true
#| message: false

## Exploring the object
rm(project_info)
rse_SRP048565
```

With the RSE object generated, I proceeded to transform the counts per nucleotide to counts per read.

```{r Transforming counts}
#| echo: true
#| eval: true

assay(rse_SRP048565, "counts") <- compute_read_counts(rse_SRP048565)

## Exploring the sample attributes
head(rse_SRP048565$sra.sample_attributes)
```

Because there is consistency in the sample attributes, I expanded them to have a more detailed view of the experimental conditions and prepared the data for the statistical modeling.

```{r Expanding sample attributes}
#| echo: true
#| eval: true
#| message: true

## As in the class example, here I used the experiment data
rse_SRP048565 <- expand_sra_attributes(rse_SRP048565)
colData(rse_SRP048565)[
  , 
  grepl("^sra_attribute", 
  colnames(colData(rse_SRP048565)))
]

## Statistical modeling preparation of my interest variables
rse_SRP048565$sra_attribute.time <- factor(
    rse_SRP048565$sra_attribute.time,
    levels = c("2hr", "6hr", "12hr")
)
rse_SRP048565$sra_attribute.infection <- factor(
    rse_SRP048565$sra_attribute.infection,
    levels = c("Control", "Afum")
)

## Summary of this variables
summary(as.data.frame(colData(rse_SRP048565)[
    , 
    grepl("^sra_attribute.(time|infection)$", 
    colnames(colData(rse_SRP048565)))
]))
```

As it is observed, there are 6 samples for each time point (2, 6, and 12 hours) and for each condition (infected and non-infected). This allows for a balanced design in the differential expression analysis. Because of the latter, time anf infection were used as the main variables of interest in the statistical modeling.

## Data Quality Assessment and Filtering

```{r Data quality assessment}
#| echo: true
#| eval: true
#| message: true

## Assessing data quality through QC
rse_SRP048565$assigned_gene_prop <- rse_SRP048565$recount_qc.gene_fc_count_all.assigned / rse_SRP048565$recount_qc.gene_fc_count_all.total
summary(rse_SRP048565$assigned_gene_prop)

## Visualizing the distribution of assigned gene proportions
hist(
  rse_SRP048565$assigned_gene_prop, 
  main = "Distribution of Assigned Gene Proportions", 
  xlab = "Proportion of Assigned Genes", 
  col = "lightblue"
)
```

The histogram of the assigned gene proportions shows that most samples have a high proportion of assigned genes, which is a **good indicator of data quality**. However, there is some variability among the samples, which could be due to differences in sequencing depth or sample quality. To further investigate this, I created a boxplot to compare the assigned gene proportions between the infected and non-infected samples across the different time points.

```{r Boxplot of assigned gene proportions by treatment}
#| echo: true
#| eval: true

boxplot(
  rse_SRP048565$assigned_gene_prop ~ rse_SRP048565$sra_attribute.infection, 
  main = "Assigned Gene Proportions by Treatment", 
  xlab = "Treatment", 
  ylab = "Proportion of Assigned Genes", 
  col = "lightgreen"
)
```

Because no evidence supported a difference in the assigned gene proportions between the infected and non-infected samples, I decided to filter the data using `edgeR::filterByExpr()` for gene expression, as the lowly expressed genes may not be biologically relevant and could introduce noise into the analysis.

```{r Library for filtering}
#| echo: true
#| eval: true
#| message: false

if (!requireNamespace("edgeR", quietly = TRUE)) {
    BiocManager::install("edgeR", update = FALSE)
}

invisible(library("edgeR"))
```
```{r Extracting count matrix and group information for filtering}
## Extracting the count matrix
counts_matrix <- assay(rse_SRP048565)

## Information about the groups
group <- as.factor(rse_SRP048565$sra_attribute.infection)

## Expression filter
keep_genes <- filterByExpr(counts_matrix, group=group)

## Fetching the filtered data
rse_SRP048565_filtered <- rse_SRP048565[keep_genes,]

## Exploring dimensions of the filtered data
dim(rse_SRP048565_filtered)
```

```{r Percentage of genes retained after filtering}
#| echo: true
#| eval: true

rm(group, keep_genes)
invisible(gc())

## Percentage of genes retained after filtering
round(nrow(rse_SRP048565_filtered) * 100 / nrow(rse_SRP048565), 2)
```

After applying the filtration function to the data, I retained approximately 51.2% of the genes, which indicates that a significant portion of the lowly expressed genes were removed from the analysis. This step is crucial for improving the statistical power of the differential expression analysis and reducing the likelihood of false positives.

## Normalization

```{r Normalization}
#| echo: true
#| eval: true

## Scaling factors for normalization
dge <- DGEList(
  counts = assay(rse_SRP048565_filtered, "counts"), 
  genes = rowData(rse_SRP048565_filtered)
)
dge <- calcNormFactors(dge)
```

Normalization of RNA-seq expression data is necessary because compositional bias and differences in library sizes can distort the results of differential expression analysis (@robinson-2010). The `calcNormFactors` function from the `edgeR` package estimates scaling factors (by default using the TMM method) to adjust for these biases between samples. These normalization factors are incorporated into the statistical model, ensuring that expression levels are comparable across samples and enabling accurate identification of differentially expressed genes.

## Differential Expression Analysis

Differential expression was assessed by comparing infected vs non-infected samples within each time point (2, 6, and 12 hours). To enable time-specific contrasts, we modeled each time-infection combination as a separate group in the design matrix. This approach provides straightforward contrasts for infected vs. control at each time point while accounting for baseline differences among time points.

```{r Contrasts for model matrix}
#| echo: true
#| eval: true

if (!requireNamespace("limma", quietly = TRUE)) {
    BiocManager::install("limma", update = FALSE)
}

library("limma")

## Fetching metadata for the model matrix
time <- droplevels(rse_SRP048565_filtered$sra_attribute.time)
infection <- droplevels(rse_SRP048565_filtered$sra_attribute.infection)

## Combined group factor of time x infection
group <- interaction(time, infection, sep = "_", drop = TRUE)
```

The design matrix was structured to include separate groups for each time-infection combination, so there was no intercept as each computed coefficient corresponds to a specific time-infection group average expression.

```{r Model matrix without intercept}
#| echo: true
#| eval: true
#| message: true

## Design matrix without intercept
mod <- model.matrix(~ 0 + group)

## Viewing the model matrix levels
colnames(mod) <- levels(group)
colnames(mod) <- paste0("tr", colnames(mod))
colnames(mod)
```

```{r Voom model}
#| echo: true
#| eval: true

vGene <- voom(dge, mod, plot = TRUE)
```

The plot shows that the samples with lower count sizes have higher variability, meaning that the genes with low expression levels tend to have more distance with the mean in their expression measurements, as they are more affected by technical noise and sampling variability.

```{r Fitting the linear model}
#| echo: true
#| eval: true

## The model matrix is already contained in the voom object, so we can directly fit the linear model
fit <- lmFit(vGene)

## Contrasts for infected vs control at each time point
cont <- makeContrasts(
  Inf_vs_Ctrl_2hr  = tr2hr_Afum  - tr2hr_Control,
  Inf_vs_Ctrl_6hr  = tr6hr_Afum  - tr6hr_Control,
  Inf_vs_Ctrl_12hr = tr12hr_Afum - tr12hr_Control,
  levels = mod
)

## Re-expressing the model fit with the specified contrasts
## Adjust the variance estimates for the contrasts
eb_results <- eBayes(contrasts.fit(fit, cont))

## Tables
tt_2hr  <- topTable(eb_results, coef="Inf_vs_Ctrl_2hr",  number=Inf)
tt_6hr  <- topTable(eb_results, coef="Inf_vs_Ctrl_6hr",  number=Inf)
tt_12hr <- topTable(eb_results, coef="Inf_vs_Ctrl_12hr", number=Inf)

dim(tt_12hr)
```

All genes are considered in each table (all have the same dimensions), sorted by their significance with an `FDR < 0.05`, this election allows me to filter according to an specified threshold.

### Time-specific DEGs

#### 2 hours
```{r DEGs at 2 hours}
#| echo: true
#| eval: true
#| message: true

## DEGs with FDR < 0.05
table(tt_2hr$adj.P.Val < 0.05)
```

```{r DEGs with adj.P.Val < 0.05 and logFC > 1}
#| echo: true
#| eval: true
#| message: true

## DEGs with FDR < 0.05
head(
  tt_2hr[(tt_2hr$adj.P.Val < 0.05) & (abs(tt_2hr$logFC) > 1), ], 
  n = 5
)
```

```{r Volcano plot for 2 hours}
#| echo: true
#| eval: true
#| message: true
#| fig-width: 8
#| fig-height: 6
#| dpi: 300

## Reorganizing the gene names with the LM results order
gene_names <- rowData(rse_SRP048565_filtered)$gene_name
names(gene_names) <- rownames(rse_SRP048565_filtered)

volcanoplot(
  eb_results,
  coef = "Inf_vs_Ctrl_2hr",
  highlight = 3,
  names = gene_names[rownames(eb_results)]
)
```

At the 2 hour time point, there are 255 DEGs with an `FDR < 0.05`, indicating an early host response to _A. fumigatus_ infection. The volcano plot shows that most of the DEGs have a log fold change (`logFC`) close to zero, suggesting that the changes in gene expression are relatively modest at this early stage of infection. However, from the three most significant DEGs two are upregulated (*ZNF750* and *ZFAND1*) and one is downregulated for the infected samples compared to controls.

[ZNF750](https://www.genecards.org/cgi-bin/carddisp.pl?gene=ZNF750&keywords=ZNF50) and [ZFAND1](https://www.genecards.org/cgi-bin/carddisp.pl?gene=ZFAND1&keywords=ZFAND1) both are transcription factors with zinc finger domains critical for regulating epithelial homeostasis and stress adaptation. The first contributes to epithelial maturation and barrier integrity. While the second regulates cytoplasmic stress granule turnover and facilitating proteasome recruitment during acute cellular stress, which supports protein quality control and recovery from exogenous insults.

In the other hand, the downregulated gene [CITED4](https://www.genecards.org/cgi-bin/carddisp.pl?gene=CITED4&keywords=CITED4) functions as a transcriptional coactivator interacting with _CBP/p300_ and _TFAP2_, modulating RNA polymerase II–dependent gene expression and influencing transcriptional programs linked to cellular differentiation and stress responses.

There is also a gene with a `logFC > 1` and an `adj.P.Val < 0.05`, which is [IL36G](https://www.genecards.org/cgi-bin/carddisp.pl?gene=IL36G&keywords=IL36G), a member of the IL-1 cytokine family, functions as a pro-inflammatory epithelial-derived cytokine that activates _NF-κB_ and _MAPK_ signaling.

In the context of _Aspergillus fumigatus conidia_ exposure in HBECs (an interaction known to **trigger inflammatory signaling, oxidative stress, and epithelial remodeling**), the upregulation of stress-response regulators such as *ZFAND1* and differentiation-associated factors like *ZNF750* may reflect **activation of epithelial defense and repair mechanisms**. Conversely, the downregulation of _CITED4_ in infected cells could indicate **suppression or reprogramming of specific transcriptional coactivation pathways during the host response**, potentially shifting the epithelial transcriptional landscape toward stress and innate immune programs.

The upregulated expression of _IL36G_ may indicate an early pro-inflammatory response to _A. fumigatus_ exposure. This suggests that the infected HBECs are **initiating an inflammatory signaling cascade** in response to the fungal _conidia_, which could contribute to the recruitment of immune cells and the activation of downstream defense mechanisms.

```{r Heatmap of the top 20 DEGs at 2 hours}
#| echo: true
#| eval: true
#| message: true

if (!requireNamespace("pheatmap", quietly = TRUE)) {
    BiocManager::install("pheatmap", update = FALSE)
}
library("pheatmap")

## Extracting the genes of interest
expression_ht <- vGene$E[rank(tt_2hr$adj.P.Val) <= 50, ]

df <- as.data.frame(
  colData(rse_SRP048565_filtered)[
    , 
    "sra_attribute.infection", 
    drop = FALSE
  ]
)
colnames(df) <- "Infection"

# (opcional pero recomendable) asegurar factor
df$Infection <- factor(df$Infection)

pheatmap(
  expression_ht,
  cluster_rows = TRUE,
  cluster_cols = TRUE,
  annotation_col = df,
  show_rownames = FALSE,
  show_colnames = FALSE,
  main = "Top 50 DEGs at 2 hours"
)
```

At the 2 hours-time point, two main clusters appear, one full of control samples and the other mixing both treatments. This might suggests that the infection samples have gene expression levels that are similar in a way to those shown in some control samples. If a Venn diagram would be made, several significant genes may be present in control and infection samples.

Still, most control samples cluster together showing low similarity with the infected ones.

#### 6 hours
```{r DEGs at 6 hours}
#| echo: true
#| eval: true
#| message: true

table(tt_6hr$adj.P.Val < 0.05)
```

```{r DEGs with adj.P.Val < 0.05 and logFC > 1 at 6 hours}
#| echo: true
#| eval: true
#| message: true

## DEGs with FDR < 0.05
head(
  tt_6hr[(tt_6hr$adj.P.Val < 0.05) & (abs(tt_6hr$logFC) > 1), ], 
  n = 5
)
```

At the 6-hour time point, there are no DEGs with an FDR < 0.05, suggesting an abscent responce to the pathogen in a more advanced stage of the illness. Because of the latter, it is possible that the host response is transient and peaks at an earlier time point, or that the response is more complex and involves changes in gene expression that are not captured by the statistical analysis at this time point.

#### 12 hours
```{r DEGs at 12 hours}
#| echo: true
#| eval: true
#| message: true

table(tt_12hr$adj.P.Val < 0.05)
```

```{r DEGs with adj.P.Val < 0.05 and logFC > 1 at 12 hours}
#| echo: true
#| eval: true
#| message: true

head(
  tt_12hr[(tt_12hr$adj.P.Val < 0.05) & (abs(tt_12hr$logFC) > 1), ], n = 5
)
```

```{r Volcano plot for 12 hours}
#| echo: true
#| eval: true
#| message: true
#| fig-width: 8
#| fig-height: 6
#| dpi: 300

volcanoplot(
  eb_results, 
  coef = "Inf_vs_Ctrl_12hr", 
  highlight = 3, 
  names = gene_names[rownames(eb_results)]
)
```

For the 12 hour time point, 4615 DEGs were identified with an `FDR < 0.05`, indicating a more robust host response to _A. fumigatus_ infection at this later stage. This time point three most significant DEGs are all upregulated in the infected samples compared to controls, an they are [SERPINB2](https://www.genecards.org/cgi-bin/carddisp.pl?gene=UBN1&keywords=UBN1), [PRDM1](https://www.genecards.org/cgi-bin/carddisp.pl?gene=MAD2L2&keywords=MAD2L2) and an unidentified gene (_AC091633_). _SERPINB2_ acts as a protease inhibitor involved in fibrinolysis and inflammatory signaling pathways, contributing to regulation of extracellular proteolytic activity and tissue remodeling. While _PRDM1_ functions as a transcriptional repressor that regulates interferon-responsive genes and immune-related transcriptional programs.

By examining the genes with a `logFC > 1` and an `adj.P.Val < 0.05`, the other two most significant DEGs are [ANGPTL4](https://www.genecards.org/Search/Keyword?queryString=ANGPTL4) and [SLC20A2](https://www.genecards.org/cgi-bin/carddisp.pl?gene=SLC20A2&keywords=SLC20A2) and also upregulated during infection. The first mediates sodium-dependent phosphate uptake, contributing to cellular metabolic balance and mineral homeostasis, processes that can influence cellular signaling and survival under stress conditions. The second is a secreted regulator of lipid metabolism and endothelial behavior, modulating vascular permeability, extracellular matrix interactions, and cellular adhesion.

In bronchial epithelial cells exposed to _Aspergillus fumigatus conidia_, these upregulated genes may reflect coordinated responses integrating metabolic adaptation (_SLC20A2_), modulation of epithelial-vascular interactions and barrier integrity (_ANGPTL4_), control of extracellular protease activity during inflammation (_SERPINB2_), and transcriptional reprogramming of immune signaling pathways (_PRDM1_). Given that fungal contact induces epithelial activation, cytokine production, oxidative stress, and tissue remodeling, modulation of these genes may contribute to **balancing host defense**, **barrier maintenance**, and **inflammatory regulation** during early antifungal responses.

```{r Heatmap of the top 20 DEGs at 12 hours}
#| echo: true
#| eval: true
#| message: true

## Extracting the genes of interest
expression_ht <- vGene$E[rank(tt_12hr$adj.P.Val) <= 50, ]

df <- as.data.frame(
  colData(rse_SRP048565_filtered)[
    , "sra_attribute.infection", 
    drop = FALSE
  ]
)
colnames(df) <- "Infection"

# (opcional pero recomendable) asegurar factor
df$Infection <- factor(df$Infection)

pheatmap(
  expression_ht,
  cluster_rows = TRUE,
  cluster_cols = TRUE,
  annotation_col = df,
  show_rownames = FALSE,
  show_colnames = FALSE,
  main = "Top 50 DEGs at 12 hours"
)
```

This time point shows a very similar behaviour to that in the 2-hours time, which could mean that the dynamics in response to the infection of _A. fumigatus conidia_ follows an equivalent expression pattern as the infection evolves.

```{r Cleaning the variables space}
## Cleaning the variables in the NameSpace
rm(list = ls())
invisible(gc())
```

# Results

_A. fumigatus conidia_ is an ambient fungus that infects its host through inhalation, and the interaction between its spores and respiratory epithelial cells is a critical step in the infectious process. The scRNA-seq revealed differentialy expressed genes in the infected samples compared to controls at 2 and 12 hours, but not at 6 hours. At 2 hours, the DEGs were mostly related to stress response and epithelial homeostasis, while at 12 hours, the DEGs were more associated with metabolic adaptation, vascular interactions, and immune regulation. These findings suggest a dynamic host response to _A. fumigatus_ infection that evolves over time, with early activation of defense mechanisms followed by more complex regulatory processes as the infection progresses.

It has been reported that HBECs participates in the attenuation of the inflammatory response to _A. fumigatus_ conidia (@richard-2018), which is consistent with the observed upregulation of genes involved in immune regulation and barrier maintenance at later time points. The absence of DEGs at 6 hours may indicate a transient phase in the host response, where the initial activation subsides before a more robust response emerges at 12 hours. Overall, these results provide insights into the temporal dynamics of the epithelial response to fungal infection and highlight potential targets for therapeutic intervention in aspergillosis.

# References
